packages/cli/src/commands/upgrade/upgrade.ts

```diff
@@ -98,10 +93,14 @@ const getPatch = async (currentVersion, newVersion, config) => {
   return patchWithRenamedProjects;
 };
 
-const getVersionToUpgradeTo = async (argv, currentVersion, projectDir) => {
+const getVersionToUpgradeTo = async (
+  argv: Array<string>,
+  currentVersion: string,
+  projectDir: string,
+) => {
   const newVersion = argv[0]
     ? semver.valid(argv[0]) ||
-      (semver.coerce(argv[0]) ? semver.coerce(argv[0]).version : null)
+      (semver.coerce(argv[0]) ? semver.coerce(argv[0])!.version : null)
```

Comment:
This looks like a bug in TS, `semver.coerce(argv[0])` is clearly not null

Reply:
Not really. You assumed that `semver.coerce` is a pure function. If it had side effects it could return null second time. I am pretty sure if you assigned `semver.coerce(argv[0])` to a variable and then reuse everything would be okay.

Reply:
Fair point